client/cli/src/params/pruning_params.rs

```diff
@@ -30,12 +30,16 @@ pub struct PruningParams {
 	/// or for all of the canonical blocks (i.e 'archive-canonical').
 	#[clap(alias = "pruning", long, value_name = "PRUNING_MODE")]
 	pub state_pruning: Option<String>,
```

Comment:
I think there needs to be separate option for this. It does not really control pruning history length, but rather how soon unfinalized branches are discared. @bkchr wdyt?